pythonforandroid/python.py

```diff
@@ -404,22 +437,28 @@ def build_arch(self, arch):
         build_dir = join(recipe_build_dir, self.build_subdir)
         ensure_dir(build_dir)
 
-        if not exists(join(build_dir, 'python')):
-            with current_directory(recipe_build_dir):
-                # Configure the build
-                with current_directory(build_dir):
-                    if not exists('config.status'):
-                        shprint(
-                            sh.Command(join(recipe_build_dir, 'configure')))
-
-                # Create the Setup file. This copying from Setup.dist
-                # seems to be the normal and expected procedure.
-                shprint(sh.cp, join('Modules', 'Setup.dist'),
-                        join(build_dir, 'Modules', 'Setup'))
-
-                shprint(sh.make, '-j', str(cpu_count()), '-C', build_dir)
-        else:
-            info('Skipping {name} ({version}) build, as it has already '
-                 'been completed'.format(name=self.name, version=self.version))
-
-        self.ctx.hostpython = join(build_dir, 'python')
+        with current_directory(recipe_build_dir):
+            # Configure the build
+            with current_directory(build_dir):
```

Comment:
Why do we switch current directory twice in a row? Is it because the second time it's a relative path?

Reply:
> Why do we switch current directory twice in a row? Is it because the second time it's a relative path?

I'm not pretty sure of this, since I only removed one indentation level, but, apart of relative paths, I think that it's because we need to write `config.status` inside `native-build` folder...also all the thing makes sense in one context because it's a part of the build process of  python.

Anyway...maybe @inclement can add some words or correct me. He did  the original **hostpython3/python3** recipes :smile: (I only refactored a little, some months back, to have the python2 version of the thing based on the work He made...but the concept and logic are the @inclement's ones :wink:)

Reply:
Thanks, I was just curious, maybe if I get to curious I'll play with it later. But let's get this PR merged first :smile: